paddle/fluid/operators/ngraph/ngraph_engine.cc

```diff
@@ -232,31 +229,32 @@ NgraphEngine::NgraphEngine(const framework::Scope& scope,
   var_node_map_ = std::make_shared<
       std::unordered_map<std::string, std::shared_ptr<ngraph::Node>>>();
 
+  std::lock_guard<std::mutex> lock(ng_mutex_);
+
+  if (!wp_backend_.lock()) {
+    try {
+      VLOG(3) << "ngraph creating CPU  backend.";
+      backend_ = ngraph::runtime::Backend::create("CPU");
```

Comment:
here, is it possible to create cpu backend for multiple times and keep multiple instances?
why not keep backend as static and keep one instance? since only ngraph function and executable are useful for next iteration execution, backend is for creating or removing.

original definition is static shared_ptr, there is no explicit releasing, so when it is released early? 

Reason to keep static shared_ptr is on SCR, backend may only have one instance, and device is used exclusively, then it is still possible to support PE, but if use multiple backends, it becomes difficult.

Reply:
@LeoZhao-Intel 

> but if use multiple backends, it becomes difficult.
This code has one backend , Baojun sharing reference (single instance) via weak_ptr to others. First execution of this part of code  is creating a backend. If all ng-engine die , local shared_ptr backend_ are destroyed and backend is deleted, when the last one die.

Reply:
if backend changed to non-static, then each NGraphEngine will have one instance, and NGraphEngine instance is created for each op kernel execution, 
```
template <typename DeviceContext, typename T>
class NgraphEngineKernel : public framework::OpKernel<T> {
 public:
  void Compute(const framework::ExecutionContext& ctx) const override {
    auto& scope = ctx.scope();
    auto place = ctx.GetPlace();

    NgraphEngine ngraph_engine(scope, place, ctx);
    ngraph_engine.Run(scope, place);
  }
};
``` 
It is possible in PE that multiple kernel instance running in parallel. I agree to use single backend, and expect to keep single backend not multiple.

Reply:
one corner case, for only one ngraphengine op, whether backend is created and destroy for each iteration?

Reply:
ok, got answer, then it will use cached ctx